src/plan.rs

```diff
@@ -0,0 +1,139 @@
+use std::{
+    cmp::Ordering,
+    collections::{BinaryHeap, HashMap},
+};
+
+pub struct Id {
+    id: usize,
+}
+
+pub struct Plan<T> {
+    pub time: f64,
+    pub data: T,
+}
+
+#[derive(PartialEq, Debug)]
+pub struct Record {
+    pub time: f64,
+    id: usize,
+}
+
+impl Eq for Record {}
+
+impl PartialOrd for Record {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for Record {
+    fn cmp(&self, other: &Self) -> Ordering {
+        let time_ordering = self.time.partial_cmp(&other.time).unwrap().reverse();
+        if time_ordering == Ordering::Equal {
+            // Break time ties in order of plan id
+            self.id.cmp(&other.id).reverse()
+        } else {
+            time_ordering
+        }
+    }
```

Comment:
I think this would be cleaner as:
```suggestion
        match time_ordering { 
            Ordering::Equal => self.id.cmp(&other.id).reverse(),
            _ => time_ordering
        }
```